### ISSUE TITLE
Correct array shape() for mx.lstm

### ISSUE THREAD

**Issue by nmatare** [dmlc/mxnet#5147]
Hello, I've been following the [RNN example ](http://dmlc.ml/rstats/2016/08/19/recurrent-models-and-examples-with-mxnet-r.html) but  I'm unable to adopt it to a lay example. Specifically, I'm unable to set the correct parameters for mx.lstm without it throwing a shape() error exception. [similar issue](http://stackoverflow.com/questions/41294025/correctly-formatting-data-for-lstm-recurrent-neural-network-in-r-mxnet)

Could someone please elaborate on the below for a non-text example:

> num.embed: integer The output dim of embedding.
> num.label: integer The number of labels.
> input.size: integer The input dim of one-hot encoding of embedding
> seq.len: integer The length of the input sequence.

## Environment 

```
sessionInfo()
R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server 7.2 (Maipo)

```
## Example

```
library(mxnet)
data(iris)
y <- as.integer(iris$Species)
x <- iris[,-5]

model <- mx.lstm(
				train.data = list(data = t(x), label = t(y)),
				ctx = mx.cpu(),
				update.period = 1,
				num.lstm.layer = 1, 
				num.hidden = 4, 			
				num.embed = 4, 
				
				seq.len = 4,
				num.label = 3,
				input.size = 1,

				num.round = 5, 
				batch.size = 10, 
				initializer = mx.init.uniform(0.1), 
				learning.rate = 0.05,
				clip_gradient = 1
)
```

## Error

`> Check failed: from.shape() == to->shape() operands shape mismatchfrom.shape = (10,1) to.shape=(10,4)
`

Thanks

**Comment by lahdeaho** [comment]
Hi,

I think you should have same size of input and output datasets when using `mx.lstm`.

```
library(mxnet)
data(iris)
y <- as.integer(iris$Species)[1:5!=5]
x <- iris[1:5!=5,-5]

input <- t(x)
output <- rbind(t(y), matrix(0,ncol=length(y), nrow=3))

test <- iris[1:5==5,]

model <- mx.lstm(
     list(data = input, label = output),
     list(data = input, label = output),
     ctx = mx.gpu(),
     update.period = 1,
     num.lstm.layer = 1, 
     num.hidden = 4, 			
     num.embed = 1, 
     
     seq.len = 4,
     num.label = 3,
     input.size = 1,
     
     num.round = 5, 
     batch.size = 4, 
     initializer = mx.init.uniform(0.1), 
     learning.rate = 0.05,
     clip_gradient = 1
)
```

`seq.len` should match row size and `batch.size` should match with columns size.

Another thing is this dataset.
I don't know what's the purpose of using lstm model for this kind of data (iris), but it's not the point here.

I hope this helps, I've also been struggling these issues with `mx.lstm` .

**Comment by thirdwing** [comment]
@nmatare In current implementation, your `data` and `label` should have the same dimensions just like what we have in the tutorial: http://dmlc.ml/rstats/2016/08/19/recurrent-models-and-examples-with-mxnet-r.html

```r
> dim(X.train.data)
[1]    32 31360
> dim(X.train.label)
[1]    32 31360
> dim(X.val.data)
[1]   32 3456
> dim(X.val.label)
[1]   32 3456
```

CC @ziyeqinghan

**Comment by chaitanyabjoshi** [comment]
This issue should not be closed as documentation (even in version 0.11) is not fixed and is so cryptic that no meaning can be made out it by reading!

**Comment by Garglesoap** [comment]
Where are these tutorials today? The links are all broken and I cannot find any simple regression/classification examples. Why does every lstm have to be character based...
